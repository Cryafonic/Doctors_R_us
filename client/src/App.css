.float-right {
  margin-left: 50%;
}

.card {
  margin-top: 6%;
  margin-left: auto;
  margin-right: auto;
  max-width: 100%;
}

.input-center {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.login-button {
  margin-left: 10.5%;
}

a {
  color: darkgray;
  text-decoration: none;
}

a:hover {
  color: gray;
  /* text-decoration: none */
}

.top-button {
  margin: 10px;
  float: right;
}

.logout-button {
  position: absolute;
  right: 10px;
}

.appointment-heading {
  margin: 1%;
}

button {
  margin: 0.5%;
}

.mobile-alert {
  margin: 1% 1% 1% 85%;
}

.widthAppointmentCard {
  width: 450px;
}

.updateAppointmentButton {
  margin: 1% 1% 3% 60%;
}

.marginAppointmentButton {
  margin: 0 0 0 65%
}  

.marginAllaround {
  margin: 1%;
}

.widthAlert {
  width: 200px;
}

.right70margin {
  margin: 0 70% 0 0;
}

.dateWidth {
  width: 300px;
}

.dateScheduledSubmitButton {
  width: 100px;
  margin: 1% 0 1% 70%;
}

.setWidth460 {
  width: 460px;
}

.diagnosisSubmitButton {
  margin: 1% 1% 1% 80%;
}

.docCard {
  width: 500px; 
  height: 1%;
}

.docH2 {
  margin: 1% 46% 1% 1%;
}

.docSet50 {
  width: 50%;
}

.docSubmitButton {
  margin: 1% 1% 1% 75%;
}

.loginCard {
  width: 30%;
   height: 1%;
}

.loginInputGroup {
  width: 80%;
}

.navImage {
  height: 35px; 
  margin: 0 0 0 10px;
}

.h2Reg {
  margin: 1% 74% 1% 1%;
}

@media only screen and (max-width: 1050px) {
  .mobile-alert {
    margin: 1% 1% 1% 58%;
  }
}
