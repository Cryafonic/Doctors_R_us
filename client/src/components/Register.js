import axios from "axios";
import React, { useState } from "react";
import { Redirect } from "react-router-dom";
import Button from "react-bootstrap/Button";
import InputGroup from "react-bootstrap/InputGroup";
import FormControl from "react-bootstrap/FormControl";
import Card from 'react-bootstrap/Card';

// if user does not have a login they can navigate to the register and create a login for themselfs.
function Register() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [name, setName] = useState("");
  const [surname, setSurename] = useState("");
  const [message, setMessage] = useState("");

  // registers the person on submit
  const onSubmit = (e) => {
    e.preventDefault();
    const register = {
      email,
      password,
      name,
      surname,
    };

    axios
      .post("http://localhost:8080/api/patient/register", register)
      .then((res) => {
        console.log(res.data);
        if (res.data.message) {
          setMessage(res.data.message);
        }
      })
      .catch((err) => {
        console.log(err.email);
      });
  };

  const hanldeChange = (e) => {
    const value = e.target.value;
    const name = e.target.name;
    if (name === "email") {
      setEmail(value);
    } else if (name === "password") {
      setPassword(value);
    } else if (name === "name") {
      setName(value);
    } else {
      setSurename(value);
    }
  };
    // checks for what page to render
  if (message === "registerd") {
    return <Redirect to="/login" />;
  } else {
    return (
      <div>
        <form onSubmit={onSubmit}>
        <Card className="input-center card bg-light docCard" >
          <h2 className='h2Reg' >Register:</h2>

          <InputGroup className="mb-2 ms-2 docSet50" >
            <FormControl
              value={email}
              onChange={hanldeChange}
              type="email"
              name="email"
              required
              placeholder="Email"
            />
        </InputGroup>
        <InputGroup className="mb-2 ms-2 docSet50" >
            <FormControl
              value={password}
              onChange={hanldeChange}
              type="password"
              name="password"
              required
              placeholder="Password"
            />
            </InputGroup>
            <InputGroup className="mb-2 ms-2 docSet50" >
            <FormControl
              value={name}
              onChange={hanldeChange}
              type="text"
              name="name"
              required
              placeholder="Name"
            />
            </InputGroup>
            <InputGroup className="mb-2 ms-2 docSet50" >
            <FormControl
              value={surname}
              onChange={hanldeChange}
              type="text"
              name="surname"
              required
              placeholder="Surname"
            />
          </InputGroup>
          <Button type="submit" className='docSubmitButton' >Register</Button>
          </Card>
        </form>
        
      </div>
    );
  }
}

export default Register;
