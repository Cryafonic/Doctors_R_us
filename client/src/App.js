import 'bootstrap/dist/css/bootstrap.min.css';
import "./App.css";
import React, {useState} from 'react';
import { HashRouter as Router, Route } from "react-router-dom";

//components 
import AppointmentList from './components/AppointmentList';
import { NavBar }  from './components/NavBar';
import AppointmentCreate from "./components/AppointmentCreate";
import AppointmentEdit from "./components/AppointmentEdit";
import Login from './components/Login';
import Register from './components/Register';
import DocRegister from './components/DocRegister';
import Diagnosis from './components/Diagnosis';
import DateScheduled from './components/DateScheduled';


export const UserContext = React.createContext([]);

function App() {
  const [user, setUser] = useState({});

  function Logout() {
    window.location = '/';
  }
  console.log(user);

  // renders the router with all routes it can to.
  return ( 
    <UserContext.Provider value={[user, setUser]} >
      <div>
        <Router>
          <NavBar logout={Logout} />
          <Route path="/" exact component={AppointmentList} />
          <Route path="/login" exact component={Login} />
          <Route path="/register" exact component={Register} />
          <Route path="/doctor/register" exact component={DocRegister} />
          <Route path="/edit/:id" exact component={AppointmentEdit} />
          <Route path="/add" exact component={AppointmentCreate} />
          <Route path="/diagnosis/:id" exact component={Diagnosis} />
          <Route path="/dateScheduled/:id" exact component={DateScheduled} />
        </Router>
      </div>
    </UserContext.Provider>
  );
}

export default App;
