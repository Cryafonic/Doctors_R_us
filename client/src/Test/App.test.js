import React from 'react';
import {render as rtlRender, screen} from '@testing-library/react'
import App from '../App';
import Register from '../components/Register';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event'
import {
  BrowserRouter as Router,
} from 'react-router-dom'

describe('Snapshot tests for main app component', () => {
  it('Main app renders correctly', () => {
    const tree = rtlRender(<App />);
    expect(tree).toMatchSnapshot();
  });

  it("testing if components match is falsy", async () => {
    const wrapper = shallow(<Register />);
    const wrapper1 = shallow(<App />);
    expect(wrapper).not.toBe(wrapper1);
  });


  it("checks if the button is working as expected", () => {
    const wrapper = shallow(<Register />);
    wrapper.find('Button').simulate('click');
  });
});

describe('Render screen elements correctly', () => {
  const render = (ui, {route = '/'} = {}) => {
    window.history.pushState({}, 'Test page', route)

    return rtlRender(ui, {wrapper: Router})
  }
  // Testing to see if our url routing works
  test('full app rendering/navigating', () => {
    render(<App />)
    expect(screen.getByText(/Appointments/i)).toBeInTheDocument();

    userEvent.click(screen.getByText(/Appointments/i));

    expect(screen.getByText(/Register page/i)).toBeInTheDocument();
  })
});
