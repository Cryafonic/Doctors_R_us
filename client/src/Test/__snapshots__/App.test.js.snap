// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for main app component Main app renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="navbar navbar-expand navbar-dark bg-dark"
        >
          <img
            class="navImage rounded"
            src="hosLogo.png"
          />
          <div
            class="navbar-nav"
          >
            <a
              class="links m-2"
              href="#/login"
            >
              Login
            </a>
            <a
              class="links m-2"
              href="#/"
            >
              Appointments
            </a>
          </div>
        </nav>
        <div>
          <div
            class="card bg-light loginCard card"
          >
            <form>
              <h2
                class="text-center"
              >
                Login
              </h2>
              <h5
                class="text-center"
              />
              <div
                class="input-center"
              >
                <div
                  class="mb-2 ms-2 loginInputGroup input-group"
                >
                  <input
                    class="form-control"
                    name="email"
                    placeholder="email"
                    required=""
                    type="email"
                    value=""
                  />
                </div>
                <div
                  class="mb-2 ms-2 loginInputGroup input-group"
                >
                  <input
                    class="form-control"
                    name="password"
                    placeholder="password"
                    required=""
                    type="password"
                    value=""
                  />
                </div>
              </div>
              <button
                class="mb-2 login-button btn btn-primary"
                type="submit"
              >
                Login
              </button>
              <a
                class="float-right"
                href="#/register"
              >
                Register page
              </a>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="navbar navbar-expand navbar-dark bg-dark"
      >
        <img
          class="navImage rounded"
          src="hosLogo.png"
        />
        <div
          class="navbar-nav"
        >
          <a
            class="links m-2"
            href="#/login"
          >
            Login
          </a>
          <a
            class="links m-2"
            href="#/"
          >
            Appointments
          </a>
        </div>
      </nav>
      <div>
        <div
          class="card bg-light loginCard card"
        >
          <form>
            <h2
              class="text-center"
            >
              Login
            </h2>
            <h5
              class="text-center"
            />
            <div
              class="input-center"
            >
              <div
                class="mb-2 ms-2 loginInputGroup input-group"
              >
                <input
                  class="form-control"
                  name="email"
                  placeholder="email"
                  required=""
                  type="email"
                  value=""
                />
              </div>
              <div
                class="mb-2 ms-2 loginInputGroup input-group"
              >
                <input
                  class="form-control"
                  name="password"
                  placeholder="password"
                  required=""
                  type="password"
                  value=""
                />
              </div>
            </div>
            <button
              class="mb-2 login-button btn btn-primary"
              type="submit"
            >
              Login
            </button>
            <a
              class="float-right"
              href="#/register"
            >
              Register page
            </a>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
